commons/src/main/java/org/sonarsource/analyzer/commons/ProfileGenerator.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * SonarQube Analyzer Commons
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.analyzer.commons;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+public class ProfileGenerator {
+
+  private static final JsonParser jsonParser = new JsonParser();
+  private static final int QUERY_PAGE_SIZE = 500;
+
+  private ProfileGenerator() {
+  }
+
+  public static File generateProfile(String serverUrl, String language, String repository, RulesConfiguration rulesConfiguration, Set<String> excludedRules) {
+    try {
+      Set<String> ruleKeys = getRuleKeys(serverUrl, language, repository);
+      ruleKeys.removeAll(excludedRules);
+      return generateProfile(language, repository, rulesConfiguration, ruleKeys);
+    } catch (IOException | XMLStreamException e) {
+      throw new RuntimeException(e);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.analyzer-commons%3Asonar-analyzer-commons-parent&branch=ruling_profile_generator&issues=AWFmxEphN21I2zDDyrbk&open=AWFmxEphN21I2zDDyrbk)

[ISSUE_KEY]: AWFmxEphN21I2zDDyrbk